src/main/scala/bazeltools/bsp4bazel/protocol/Models.scala

```diff
@@ -307,47 +314,53 @@ object SourceItemKind:
   given Encoder[SourceItemKind] =
     Encoder.instance(_.id.asJson)
 
+  given Decoder[SourceItemKind] =
+    Decoder[Int].map {
+      case 1 => SourceItemKind.File
+      case 2 => SourceItemKind.Directory
```

Comment:
We should use emap or whatever it is here can explicitly handle all cases rather than let the MatchError bubble up.